spec/std/yaml/from_yaml_spec.cr

```diff
@@ -0,0 +1,395 @@
+      {{type.id}}.from_yaml({{type.id}}.new("1.0").to_yaml).should eq {{type.id}}.new("1.0")
+    end
+
+    it "should parse a number into an {{type.id}}" do
+      {{type.id}}.from_yaml("1").should eq {{type.id}}.new("1")
+    end
+
+    it "should parse a float into an {{type.id}}" do
+      {{type.id}}.from_yaml("1.1").should eq {{type.id}}.new("1.1")
+    end
+
+    it "should raise if quoted with double quotes" do
+      expect_raises(YAML::ParseException) do
+        {{type.id}}.from_yaml "\"1.1\""
+      end
+    end
+
+    it "should raise if quoted with single quotes" do
+      expect_raises(YAML::ParseException) do
+        {{type.id}}.from_yaml "'1.1'"
+      end
+    end
+
+    it "should raise when not a float or number" do
+      expect_raises(YAML::ParseException) do
+        {{type.id}}.from_yaml("hi")
+      end
+    end
+  end
+{% end %}
+
+describe "String.from_yaml" do
+  it "should serialize and deserialize" do
+    String.from_yaml("hi".to_yaml).should eq "hi"
+  end
+
+  it "should parse a string" do
+    String.from_yaml("hello").should eq "hello"
+  end
+
+  it "should parse a single quoted string" do
+    String.from_yaml("'hello'").should eq "hello"
+  end
+
+  it "should parse a double quoted string" do
+    String.from_yaml("\"hello\"").should eq "hello"
+  end
+
+  it "should parse a literal string" do
+    String.from_yaml("|\n  hello").should eq "hello"
+  end
+
+  it "should parse a folded string" do
+    String.from_yaml(">\n  hello").should eq "hello"
+  end
+
+  context "bool values" do
+    values = [
+      "true", "True", "TRUE", "on", "On", "ON", "y", "Y", "yes", "Yes", "YES",
+      "false", "False", "FALSE", "off", "Off", "OFF", "n", "N", "no", "No", "NO",
+    ]
+    values.each do |value|
+      it "should raise if #{value}" do
+        expect_raises(YAML::ParseException) do
+          String.from_yaml(value)
+        end
+      end
+
+      it "should parse if #{value} is quoted with double quotes" do
+        String.from_yaml("\"#{value}\"").should eq value
+      end
+
+      it "should parse if #{value} is quoted with single quotes" do
+        String.from_yaml("'#{value}'").should eq value
+      end
+    end
+  end
+
+  context "nil values" do
+    values = ["", "NULL", "Null", "null", "~"]
+    values.each do |value|
+      it "should raise if #{value}" do
+        expect_raises(YAML::ParseException) do
+          # Test will an array since a standalone empty value shows as STREAM_END
+          Array(String).from_yaml("- " + value)
+        end
+      end
+
+      it "should parse if #{value} is quoted with double quotes" do
+        String.from_yaml("\"#{value}\"").should eq value
+      end
+
+      it "should parse if #{value} is quoted with single quotes" do
+        String.from_yaml("'#{value}'").should eq value
+      end
+    end
+  end
+
+  context "integers" do
+    it "should raise if an int is passed" do
+      expect_raises(YAML::ParseException) do
+        String.from_yaml("1")
```

Comment:
Actually `String.from_yaml("1")` should pass, since a non-quoted `1` *can* be a plain string.

A string doesn't depend on the scalar style (plain, quoted, ...).